packages/@aws-cdk/aws-lambda/lib/function-base.ts

```diff
@@ -290,6 +296,13 @@ export abstract class FunctionBase extends Resource implements IFunction {
     source.bind(this);
   }
 
+  public configureAsyncInvoke(options: EventInvokeConfigOptions): void {
+    new EventInvokeConfig(this, 'EventInvokeConfig', {
+      function: this,
+      ...options
+    });
+  }
+
```

Comment:
> We are now protected from multiple AWS::Lambda::EventInvokeConfig by the Construct id

I like that the constructid is how we protect against multiple EventInvokeConfig but the error messaging is not great.

My suggestion -

```suggestion
  public configureAsyncInvoke(options: EventInvokeConfigOptions): void {
    if (this.node.tryFindChild('EventInvokeConfig') !== undefined) {
      throw new Error(`An EventInvokeConfig has already been configured for the qualified function ${this.functionName}:$LATEST`);
    }

    new EventInvokeConfig(this, 'EventInvokeConfig', {
      function: this,
      ...options
    });
  }
```

and a corresponding unit test in `test.lambda.ts`

Reply:
and `this.functionName` is also a token.